gtsam/navigation/GroupEKF.h

```diff
@@ -0,0 +1,196 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+ /**
+  * @file  GroupEKF.h
+  * @brief   Extended Kalman Filter derived class for Lie groups G.
+  *
+  * This file defines the GroupEKF class template, inheriting from ManifoldEKF,
+  * for performing EKF steps specifically on states residing in a Lie group.
+  * It provides predict methods utilizing group composition, tangent space
+  * controls (via exponential map), and state-dependent dynamics functions.
+  *
+  * @date  April 24, 2025
+  * @authors Scott Baker, Matt Kielo, Frank Dellaert
+  */
+
+#pragma once
+
+#include <gtsam/navigation/ManifoldEKF.h> // Include the base class
+#include <gtsam/base/Lie.h> // Include for Lie group traits and operations
+
+#include <Eigen/Dense>
+#include <type_traits>
+#include <functional> // For std::function
+
+namespace gtsam {
+
+  /**
+   * @class GroupEKF
+   * @brief Extended Kalman Filter on a Lie group G, derived from ManifoldEKF
+   *
+   * @tparam G Lie group type providing group operations and Expmap/AdjointMap.
+   *       Must satisfy LieGroup concept (`gtsam::IsLieGroup<G>::value`).
+   *
+   * This filter specializes ManifoldEKF for Lie groups, offering convenient
+   * prediction methods based on group composition or dynamics functions defining
+   * motion in the tangent space.
+   */
+  template <typename G>
+  class GroupEKF : public ManifoldEKF<G> {
+  public:
+    using Base = ManifoldEKF<G>; ///< Base class type
+    static constexpr int n = Base::n; ///< Group dimension (tangent space dimension)
+    using TangentVector = typename Base::TangentVector; ///< Tangent vector type for the group G
+    using MatrixN = typename Base::MatrixN; ///< Square matrix of size n for covariance and Jacobians
+    using Jacobian = Eigen::Matrix<double, n, n>; ///< Jacobian matrix type specific to the group G
+
+    /// Constructor: initialize with state and covariance
+    GroupEKF(const G& X0, const MatrixN& P0) : Base(X0, P0) {
+      static_assert(IsLieGroup<G>::value, "Template parameter G must be a GTSAM Lie Group");
+    }
+
+    /**
```

Comment:
I am thinking of *moving* these to InvariantEKF and comment that, if the dynamics does not depend on the state, they should use an InvariantEKF

Reply:
I opted for
```mermaid
graph TD; 
ManifoldEKF --> LieGroupEKF --> InvariantEKF
````